src/drivers/intel/cavs/alh.c

```diff
@@ -64,15 +74,31 @@ static int alh_context_restore(struct dai *dai)
 
 static int alh_probe(struct dai *dai)
 {
+	struct alh_pdata *alh;
+
 	dai_info(dai, "alh_probe()");
 
+	if (dai_get_drvdata(dai))
+		return -EEXIST;
+
+	alh = rzalloc(SOF_MEM_ZONE_SYS_RUNTIME, SOF_MEM_FLAG_SHARED,
```

Comment:
SHARED memory alias seems to be not required here. There is only 1 dma channel attachable to a single alh dai device. Pls fix and also update `struct dai_ops::probe()` documentation to clarify when SHARED is required and when it is not necessary.